WireAPI/Sources/WireAPI/Network/NetworkService/NetworkService.swift

```diff
@@ -122,27 +125,26 @@ extension NetworkService: URLSessionDataDelegate {
     public func urlSession(
         _ session: URLSession,
         task: URLSessionTask,
-        didReceive challenge: URLAuthenticationChallenge,
-        completionHandler: @escaping (URLSession.AuthChallengeDisposition, URLCredential?) -> Void
-    ) {
-        print("task did receive challenge")
-
+        didReceive challenge:
+        URLAuthenticationChallenge
+    ) async -> (URLSession.AuthChallengeDisposition, URLCredential?) {
         let protectionSpace = challenge.protectionSpace
 
-        guard protectionSpace.authenticationMethod == NSURLAuthenticationMethodServerTrust else {
-            completionHandler(.performDefaultHandling, challenge.proposedCredential)
-            return
-        }
-
-        guard
-            protectionSpace.serverTrust != nil,
-            true // TODO: [WPB-10450] support certificate pinning
-        else {
-            completionHandler(.cancelAuthenticationChallenge, nil)
-            return
+        if protectionSpace.authenticationMethod == NSURLAuthenticationMethodServerTrust {
+            guard let trust = challenge.protectionSpace.serverTrust else {
+                // If this is missing it is Apple breaking its API contract so crash.
+                fatalError("Missing server trust")
+            }
```

Comment:
I would chain the conditions since we **must** have a `serverTrust` value when the authentication method is NSURLAuthenticationMethodServerTrust. And fallback to the else statement if we don't (which should not happen unless Apple API breaks). Or we **actually** want the app to crash if their API breaks ?

```
if protectionSpace.authenticationMethod == NSURLAuthenticationMethodServerTrust, 
    let trust = challenge.protectionSpace.serverTrust { ... } else { ... }
```

Reply:
Personally I think we want the app to crash because it is a security issue. If the method is `NSURLAuthenticationMethodServerTrust` we must validate - we shouldn't allow the request to continue. 

In our current implementation we would crash and this doesn't seem to be a problem so I'm just choosing to keep that behavior.

Reply:
makes sense 👍